### PR TITLE
udev-rules: extend tape and tape changer rules

### PR DESCRIPTION
**Pull request by joergsteffens**
This contains to extensions for the persistent name rules for tape and tape changes.

The existing rules don't work well with Amazon Webservice Storage Gateway Virtual Tape Library (VTL),
as all devices (10 tape devices, 1 tape changer) have the same ENV{ID_SERIAL}.
This leads to having only one link in /dev/tape/by-id/: the last device.

Patch 7f8ddf9 makes tape changer available by /dev/tape/by-path/.
With this, all devices are accessible by /dev/tape/by-path/.

Patch fc95dbc covers the fact, that all devices have the identical ENV{ID_SERIAL}, but additional have ENV{ID_SCSI_SERIAL}. So when ENV{ID_SCSI_SERIAL} is set, it additionally creates a link /dev/tape/by-id/ENV{ID_SCSI_SERIAL}.

Both patches do **not** cover the fact, that a /dev/tape/by-id/ENV{ID_SERIAL} link is created and overwritten by the next device, as changing this behavior might lead to inconsistencies when upgrading udev.

### CODE REVIEW

File: rules/60-persistent-storage-tape.rules [hunk 1]
```diff
@@ -9,6 +9,11 @@ ENV{UDEV_DISABLE_PERSISTENT_STORAGE_RULES_FLAG}=="1", GOTO="persistent_storage_t
 SUBSYSTEM=="scsi_generic", SUBSYSTEMS=="scsi", ATTRS{type}=="8", IMPORT{program}="scsi_id --sg-version=3 --export --whitelisted -d $devnode", \
   SYMLINK+="tape/by-id/scsi-$env{ID_SERIAL}"
 
+# iSCSI devices from the same host have all the same ID_SERIAL,
+# but additional a property named ID_SCSI_SERIAL.
```
Comment (keszybz) [inline rules/60-persistent-storage-tape.rules]: additional*ly*

**Comment by keszybz** [comment]
I'll restart amd64.

**Comment by haraldh** [comment]
@hreinecke could you please comment on this PR?

**Comment by joergsteffens** [comment]
I fixed the remarked typo.
Is there anything else I can do to get this change approved?

**Comment by joergsteffens** [comment]
Is there anything I can do to get this integrated?

**Comment by poettering** [comment]
@joergsteffens: get @hreinecke or @kaysievers to review this, they need to comment on this

**Comment by hreinecke** [comment]
Please note that we're working on replacing scsi_id with calls to sg_inq et al, so I'd rather have these changes integrated there.

**Comment by keszybz** [comment]
This only adds more `tape/by-id/*-$env{ID_SCSI_SERIAL}*` symlinks and should be fully backwards compatible, and forwards too — it shouldn't be an issue to add some different symlinks later on. Since it fixes the problem now, I don't see why this shouldn't be merged.

> Please note that we're working on replacing scsi_id with calls to sg_inq et al, so I'd rather have these changes integrated there.

This change doesn't block or interfere with that.

**Comment by poettering** [comment]
OK, I merged this now. I am not sure I follow this properly, but given that it doesn't look totally wrong, @keszybz likes it and this is unlikely to break everybody's machines, sounds OK to add.

**Pull request merged**